src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/UserEndpoint.java

```diff
@@ -0,0 +1,185 @@
+package uk.gov.ons.ctp.integration.contactcentresvc.endpoint;
+
+import static uk.gov.ons.ctp.common.log.ScopedStructuredArguments.kv;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.validation.Valid;
+import javax.validation.constraints.Email;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.ons.ctp.common.domain.SurveyType;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.common.error.CTPException.Fault;
+import uk.gov.ons.ctp.integration.contactcentresvc.UserIdentityContext;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.PermissionType;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.RoleDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.UserDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.security.UserIdentityHelper;
+import uk.gov.ons.ctp.integration.contactcentresvc.service.impl.UserService;
+
+@Slf4j
+@RestController
+@Validated
+@RequestMapping(value = "/users", produces = "application/json")
+public class UserEndpoint {
+
+  private UserIdentityHelper identityHelper;
+  private UserService userService;
+
+  @Autowired
+  public UserEndpoint(final UserIdentityHelper identityHelper, final UserService userService) {
+    this.identityHelper = identityHelper;
+    this.userService = userService;
+  }
+
+  @GetMapping("/{userName}")
+  public ResponseEntity<UserDTO> getUserByName(
+      @PathVariable(value = "userName") @Valid @Email String userName) throws CTPException {
+
+    log.info("Entering getUserByName", kv("userName", userName));
+    identityHelper.assertUserPermission(PermissionType.READ_USER);
+
+    return ResponseEntity.ok(userService.getUser(userName));
+  }
+
+  @GetMapping("/permissions")
+  public ResponseEntity<Set<PermissionType>> getLoggedInUsersPermissions() throws CTPException {
+
+    String userName = UserIdentityContext.get();
+    log.info("Entering getLoggedInUsersPermissions", kv("userName", userName));
+
+    identityHelper.assertUserValidAndActive();
+
+    // All users need to access this so no permission assertion
+    List<RoleDTO> usersRoles = userService.getUsersRoles(userName);
+    if (usersRoles.isEmpty()) {
```

Comment:
As discussed, user is not ready if no roles, and exception makes it clear to the UI and user
Otherwise UI has to observe no roles with no permissions, and UI could run into odd state with some things still shown/actionable